### DISCUSSION TITLE
How to generate bookmarks based on file names when merging PDFs？

### DISCUSSION THREAD

**Discussion by tyf2018** [qpdf/qpdf#1199] [Q&A]
I am using the following command to merge PDFs in Total Commander, and the experience is good. Now I would like to add a requirement: generate bookmarks based on file names when merging PDFs. Could you please tell me how I should modify the command?

`param=--empty --pages %P%S -- "%O".pdf`

Best regards.

**Comment by jberkenbilt** [comment]
This is not something qpdf can do at this time. I have big plans on supporting features like this, but life has taken a turn, and the chances of my getting to it in the foreseeable future are quite low. I am adding this issue and  your use case to the list of things that should work as, while it is very similar to other cases, the idea of prefixing outline hierarchies (or generating them) based on the pages is new.
